add.go

```diff
@@ -322,9 +322,9 @@ func (b *Builder) addHelper(excludes *fileutils.PatternMatcher, extract bool, de
 					if err != nil {
 						return errors.Wrapf(err, "error checking if %s is an excluded path", path)
 					}
-					// Skip the whole directory if the pattern matches exclusively
+					// Skip the directory if the pattern matches
 					if res.Excludes() == 0 && res.Matches() == 1 && info.IsDir() {
-						return filepath.SkipDir
+						return nil
 					}
```

Comment:
If `res.Excludes()` returns `0`, and `res.Matches()` returns `1`, then `res.IsMatched()` will return `true`.  If this block returns `nil`, it becomes redundant, so it can be removed.

Reply:
Fixed